test/pkg/environment/common/expectations.go

```diff
@@ -72,6 +72,9 @@ func (env *Environment) ExpectUpdated(objects ...client.Object) {
 		Eventually(func(g Gomega) {
 			current := o.DeepCopyObject().(client.Object)
 			g.Expect(env.Client.Get(env.Context, client.ObjectKeyFromObject(current), current)).To(Succeed())
+			if current.GetResourceVersion() != o.GetResourceVersion() {
```

Comment:
Discussed offline: Going to add a function comment here to describe when you should use this function so that you don't race with other controllers _also_ trying to make updates